### PR TITLE
EWPP-2965: Drupal 10 compatibility for 2.x branch.

### PR DESCRIPTION
**Pull request by brummbar**

### CODE REVIEW

File: modules/oe_translation_epoetry/modules/oe_translation_epoetry_mock/src/Controller/MockController.php [hunk 1]
```diff
@@ -39,6 +40,13 @@ class MockController extends ControllerBase {
    */
   protected $classResolver;
 
+  /**
+   * The module extension list service.
+   *
+   * @var \Drupal\Core\Extension\ModuleExtensionList
+   */
+  protected ModuleExtensionList $moduleExtensionList;
```
Comment (upchuk) [inline modules/oe_translation_epoetry/modules/oe_translation_epoetry_mock/src/Controller/MockController.php]: We don't have this type of type hint anywhere

File: modules/oe_translation_epoetry/modules/oe_translation_epoetry_mock/oe_translation_epoetry_mock.services.yml [hunk 2]
```diff
@@ -5,7 +5,8 @@ services:
     tags:
       - { name: http_client_middleware }
   oe_translation_epoetry_mock.logger.mail_logger:
```
Comment (upchuk) [inline modules/oe_translation_epoetry/modules/oe_translation_epoetry_mock/oe_translation_epoetry_mock.services.yml]: mail_logger is a typo. It should be mock_logger.

File: src/Form/TranslationRequestForm.php [hunk 3]
```diff
@@ -4,14 +4,56 @@
...
  * Form handler for the translation request entity add/edit forms.
  */
 class TranslationRequestForm extends ContentEntityForm {
 
+  /**
+   * The renderer service.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected RendererInterface $renderer;
```
Comment (upchuk) [inline src/Form/TranslationRequestForm.php]: We don't have type hinting here

File: src/Form/TranslationRequestForm.php [hunk 4]
```diff
@@ -21,7 +63,7 @@ public function save(array $form, FormStateInterface $form_state) {
     $link = $entity->toLink($this->t('View'))->toRenderable();
 
     $message_arguments = ['%label' => $this->entity->label()];
-    $logger_arguments = $message_arguments + ['link' => render($link)];
+    $logger_arguments = $message_arguments + ['link' => $this->renderer->render($link)];
```
Comment (upchuk) [inline src/Form/TranslationRequestForm.php]: This log is really not important. Better we don't render a link and keep the log simpler than to inject Renderer into aall the classes that extend this one. This was boilerplate/generated code.

**Review by upchuk — CHANGES_REQUESTED** [review]

**Review by upchuk — DISMISSED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**